python/ray/tune/tune.py

```diff
@@ -439,11 +446,25 @@ def run(
         from ray.tune.schedulers import create_scheduler
         scheduler = create_scheduler(scheduler)
 
-    if issubclass(type(search_alg), Searcher):
-        search_alg = SearchGenerator(search_alg)
-
     if not search_alg:
-        search_alg = BasicVariantGenerator()
+        search_alg = BasicVariantGenerator(
+            max_concurrent=max_concurrent_trials or 0)
+    else:
+        if isinstance(search_alg, ConcurrencyLimiter):
+            logger.warning(
+                "You have specified `max_concurrent_trials="
+                f"{max_concurrent_trials}`, but the `search_alg` is already "
+                "a `ConcurrencyLimiter`. `max_concurrent_trials` will "
+                "be ignored.")
+        elif max_concurrent_trials:
+            if max_concurrent_trials < 0:
+                raise ValueError(
+                    "`max_concurrent_trials` must be non-negative, "
+                    f"got {max_concurrent_trials}.")
```

Comment:
```suggestion
            if max_concurrent_trials < 1:
                raise ValueError(
                    "`max_concurrent_trials` must be greater or equal than 1, "
                    f"got {max_concurrent_trials}.")
```

Reply:
I've left in 0 because it stands for "no limit" in `BasicVariantGenerator`

Reply:
Though that will still work, thanks

Reply:
Ah but it won't work for the ConcurrencyLimiter, right?

Reply:
Ah I see you resolved this.